kernel/malloc.c

```diff
@@ -69,6 +69,10 @@ void solo5_free(void *) __attribute__ ((alias ("free")));
 void *solo5_calloc(size_t, size_t) __attribute__ ((alias ("calloc")));
 void *solo5_realloc(void *, size_t) __attribute__ ((alias ("realloc")));
 
+/* disable null-pointer-arithmetic warning on clang */
+#pragma GCC diagnostic ignored "-Wunknown-pragmas"
+#pragma clang diagnostic ignored "-Wnull-pointer-arithmetic"
```

Comment:
You should be able to stick this inside an `#if defined(__clang__)` block, thus avoiding the need to disable unknown pragmas on GCC.

Reply:
done